lib/phoenix/socket.ex

```diff
@@ -456,6 +456,7 @@ defmodule Phoenix.Socket do
 
   defp result({:ok, _}), do: :ok
   defp result(:error), do: :error
```

Comment:
We should be able to remove this clause if we rewrite `:error` to `{:error, :undefined}`.